### PR TITLE
#646 - MapOf and MapEnvelope do not have pretty toString method

### PR DESCRIPTION
**Pull request by SharplEr**
Done #646

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#668](https://github.com/yegor256/cactoos/issues/668) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#668](https://github.com/yegor256/cactoos/issues/668) is assigned to @amihaiemil[/z](https://www.0crat.com/u/amihaiemil), here is [why](https://www.0crat.com/footprint/C63314D6Z/471924654). The budget is 15 minutes, see [§4](http://datum.zerocracy.com/pages/policy.html#4). Please, read [§27](http://datum.zerocracy.com/pages/policy.html#27) and when you decide to accept the changes, inform @yegor256[/z](https://www.0crat.com/u/yegor256) (the architect) right in this ticket. If you decide that this PR should not be accepted ever, also inform the architect.

**Comment by 0crat** [comment]
Job [#668](https://github.com/yegor256/cactoos/issues/668) is already in scope

File: src/test/java/org/cactoos/map/MapOfTest.java [hunk 1]
```diff
@@ -108,4 +108,15 @@ public void createsMapWithFunctions() {
...
 
+    @Test
+    public void string() {
+        MatcherAssert.assertThat(
+            "Can't convert to string",
+            new MapOf<Integer, Integer>(
+                new MapEntry<>(-1, 0),
+                new MapEntry<>(1, 2)
+            ).toString(),
+            Matchers.equalTo("-1=0, 1=2")
```
Comment (amihaiemil) [inline src/test/java/org/cactoos/map/MapOfTest.java]: @SharplEr Shouldn't it be ``{-1=0, 1=2}``?

File: src/main/java/org/cactoos/map/MapEnvelope.java [hunk 2]
```diff
@@ -131,4 +132,8 @@ public final void clear() {
         return this.map.value().entrySet();
     }
 
+    @Override
```
Comment (amihaiemil) [inline src/main/java/org/cactoos/map/MapEnvelope.java]: @SharplEr what about ``MapOf``? Does it inherit ``toString()`` from this ``MapEnvelope``?

File: src/test/java/org/cactoos/map/MapOfTest.java [hunk 3]
```diff
@@ -108,4 +108,15 @@ public void createsMapWithFunctions() {
         );
     }
 
+    @Test
```
Comment (amihaiemil) [inline src/test/java/org/cactoos/map/MapOfTest.java]: @SharplEr Can you also add 2 more tests? One where there is a null or an empty string in one of the values and one where you have complex types such as a List or another Map in one of the keys or values?

File: src/test/java/org/cactoos/map/MapOfTest.java [hunk 4]
```diff
@@ -108,4 +108,15 @@ public void createsMapWithFunctions() {
...
+    public void string() {
```
Comment (amihaiemil) [inline src/test/java/org/cactoos/map/MapOfTest.java]: @SharplEr Can you rename this one: "integersToString"?

File: src/main/java/org/cactoos/map/MapEnvelope.java (hunk 2 above)
Comment (SharplEr) [reply src/main/java/org/cactoos/map/MapEnvelope.java]: Yes, it is. I use this `toString()` in the test for MapOf for example.

File: src/test/java/org/cactoos/map/MapOfTest.java (hunk 3 above)
Comment (SharplEr) [reply src/test/java/org/cactoos/map/MapOfTest.java]: Done.

File: src/test/java/org/cactoos/map/MapOfTest.java (hunk 4 above)
Comment (SharplEr) [reply src/test/java/org/cactoos/map/MapOfTest.java]: Done

File: src/test/java/org/cactoos/map/MapOfTest.java (hunk 1 above)
Comment (SharplEr) [reply src/test/java/org/cactoos/map/MapOfTest.java]: Done

**Comment by SharplEr** [comment]
@amihaiemil  done, please look again.

**Comment by amihaiemil** [comment]
@rultor merge it

**Comment by amihaiemil** [comment]
@yegor256 ping

**Comment by yegor256** [comment]
@rultor merge

**Comment by 0crat** [comment]
@ypshenychka[/z](https://www.0crat.com/u/ypshenychka) please review this job, as in [§30](http://datum.zerocracy.com/pages/policy.html#30)

**Comment by ypshenychka** [comment]
@SharplEr According to our [QA Rules](http://datum.zerocracy.com/pages/policy.html#42):
>Pull request description explains the solution proposed and contains a link to the original ticket it is related to.

Please update the description with more details on what was done in scope od the PR.

**Comment by ypshenychka** [comment]
@SharplEr According to our [QA Rules](http://datum.zerocracy.com/pages/policy.html#42):
>Messages in a ticket always start with a name of a user they are addressed to.

Please correct your code review comments by indicating an addressee in the beginning ([here](https://github.com/yegor256/cactoos/pull/668#discussion_r168923656), [here](https://github.com/yegor256/cactoos/pull/668#discussion_r168918743), [here](https://github.com/yegor256/cactoos/pull/668#discussion_r168923622) and [here](https://github.com/yegor256/cactoos/pull/668#discussion_r168923642))

**Comment by amihaiemil** [comment]
@SharplEr ping

**Comment by 0crat** [comment]
@amihaiemil[/z](https://www.0crat.com/u/amihaiemil) this job was assigned to you 5 days ago. It will be taken away from you soon, unless you close it, see [§8](http://datum.zerocracy.com/pages/policy.html#8). Read [this](http://www.yegor256.com/2014/04/13/no-obligations-principle.html) and [this](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html), please.

**Comment by amihaiemil** [comment]
@SharplEr ping

**Comment by SharplEr** [comment]
@amihaiemil @ypshenychka Done )

**Comment by ypshenychka** [comment]
@SharplEr Thanks. What about adding addressee to your code review comments? Please correct those messages as I asked [here](https://github.com/yegor256/cactoos/pull/668#issuecomment-366809875).

**Comment by amihaiemil** [comment]
@SharplEr can you also address all your comments from this PR to me? They should start with ``@amihaiemil`` -- this is also a part of the QA policy, each comment should be addressed to  only one person, to avoid noise. The idea is: if more people take part in a ticket, it is hard to track the conversation if there is no addresse. And if there are more addressees, then it's the same as writing an email with many people in CC: most of them will not reply, thinking that someone else replied and answered your question already.

**Comment by ypshenychka** [comment]
@SharplEr ping

**Comment by SharplEr** [comment]
@amihaiemil @ypshenychka  done.

**Comment by ypshenychka** [comment]
@SharplEr thanks

**Comment by ypshenychka** [comment]
@0crat quality acceptable

**Comment by 0crat** [comment]
`QA` review completed: +8 points just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka), total is [+670](https://www.0crat.com/u/ypshenychka)

**Comment by 0crat** [comment]
Order was finished, quality was "acceptable": +15 points just awarded to @amihaiemil[/z](https://www.0crat.com/u/amihaiemil), total is [+960](https://www.0crat.com/u/amihaiemil)

**Comment by 0crat** [comment]
> @0crat quality acceptable ([here](https://github.com/yegor256/cactoos/issues/668#issuecomment-368033199))

@ypshenychka You can't do that, unless you have one of these roles: `ARC`, `PO`. Your current roles are: `QA`.

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +15 points just awarded to @yegor256[/z](https://www.0crat.com/u/yegor256), total is [+10834](https://www.0crat.com/u/yegor256)

**Comment by 0crat** [comment]
This pull request [#668](https://github.com/yegor256/cactoos/issues/668) is assigned to @neonailol[/z](https://www.0crat.com/u/neonailol), here is [why](https://www.0crat.com/footprint/C63314D6Z/591762215). The budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4). Please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @yegor256[/z](https://www.0crat.com/u/yegor256) (the architect) right in this ticket. If you decide that this PR should not be accepted ever, also inform the architect.

**Comment by neonailol** [comment]
@yegor256 this job was assigned to me by mistake

**Comment by yegor256** [comment]
@0crat out

**Comment by 0crat** [comment]
> @0crat out ([here](https://github.com/yegor256/cactoos/issues/668#issuecomment-370393473))

@yegor256 @neonailol[/z](https://www.0crat.com/u/neonailol) resigned from [#668](https://github.com/yegor256/cactoos/issues/668), since the job is not in scope anymore

**Comment by 0crat** [comment]
> @0crat out ([here](https://github.com/yegor256/cactoos/issues/668#issuecomment-370393473))

@yegor256 The job [#668](https://github.com/yegor256/cactoos/issues/668) is now out of scope
